raiden/assetmanager.py

```diff
@@ -181,59 +188,135 @@ def register_channel_for_hashlock(self, channel, hashlock):
         if channel not in channels_registered:
             channels_registered.append(channel)
 
+    def register_secret(self, secret):
+        """ Register the secret with the interested channels.
+
+        Note:
+            The channel needs to be registered with
+            `register_channel_for_hashlock`.
+        """
+        hashlock = sha3(secret)
+        revealsecret_message = RevealSecret(secret)
+        self.raiden.sign(revealsecret_message)
+
+        channels_list = self.hashlock_channel[hashlock]
+        for channel in channels_list:
+            channel.register_secret(secret)
+
+            # This will potentially be executed multiple times and could suffer
+            # from amplification, the protocol will ignore messages that were
+            # already registered and sent it only until a first Ack is
+            # received.
+            # self.raiden.send_async(
```

Comment:
I tried to contain the changes to just adding the RevealSecret and splitting the unlock/withdraw methods. Sending the RevealSecret would need changed to the protocol and tasks, that I left to the PR #161